src/test/java/org/sasanlabs/service/vulnerability/openRedirect/Http3xxStatusCodeBasedInjectionTest.java

```diff
@@ -0,0 +1,658 @@
+        String redirectUrl = "/somedomain.com";
+        RequestEntity<String> requestEntity =
+                new RequestEntity<>(
+                        HttpMethod.GET, new URI("https://somedomain.com?returnTo=/somedomain.com"));
+        ResponseEntity<?> responseEntity =
+                http3xxStatusCodeBasedInjection.getVulnerablePayloadLevel7(
+                        requestEntity, redirectUrl);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.FOUND);
+        assertThat(responseEntity.getHeaders().get(LOCATION_HEADER_KEY))
+                .contains("https://somedomain.com/somedomain.com");
+        verify(http3xxStatusCodeBasedInjection, times(1))
+                .getVulnerablePayloadLevel7(requestEntity, "/somedomain.com");
+    }
+
+    @Test
+    @DisplayName(
+            "Level 8- test that the returnTo query parameter's value is not added to the Location header when it is not in the list of whitelisted URLS")
+    void
+            test_That_ReturnToQueryParameterValue_IsNotAddedToLocationHeader_WhenItIsNotInTheListOfWhiteListedURLS_Level8()
+                    throws URISyntaxException {
+        // /somedomain.com is not in the list of whitelisted urls
+        String redirectUrl = "/somedomain.com";
+        RequestEntity<String> requestEntity =
+                new RequestEntity<>(
+                        HttpMethod.GET, new URI("https://somedomain.com?returnTo=/somedomain.com"));
+        ResponseEntity<?> responseEntity =
+                http3xxStatusCodeBasedInjection.getVulnerablePayloadLevel8(
+                        requestEntity, redirectUrl);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(responseEntity.getHeaders().get(LOCATION_HEADER_KEY)).isEqualTo(null);
+        verify(http3xxStatusCodeBasedInjection, times(1))
+                .getVulnerablePayloadLevel8(requestEntity, "/somedomain.com");
+    }
+
+    @Test
+    @DisplayName(
+            "Level 8- test that the returnTo query parameter's value is not added to the Location header when it is not in the list of whitelisted URLS")
+    void
+            test_That_ReturnToQueryParameterValue_IsAddedToLocationHeader_WhenItIsInTheListOfWhiteListedURLS_Level8()
+                    throws URISyntaxException {
+        String redirectUrl = "/";
+        RequestEntity<String> requestEntity =
+                new RequestEntity<>(
+                        HttpMethod.GET, new URI("https://somedomain.com?returnTo=/somedomain.com"));
+        ResponseEntity<?> responseEntity =
+                http3xxStatusCodeBasedInjection.getVulnerablePayloadLevel8(
+                        requestEntity, redirectUrl);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.FOUND);
+        assertThat(responseEntity.getHeaders().get(LOCATION_HEADER_KEY)).contains("/");
+        verify(http3xxStatusCodeBasedInjection, times(1))
```

Comment:
as we have called this method at line 651 so this should be giving 1 time invocation. we can ignore this check.